### PR TITLE
Facet errors

### PR DESCRIPTION
**Pull request by amitjha21**
This PR addressed #1332 
It has made changes to `common/errros.js` to catch these newly created errors (`InvalidFacetSorting` and `InvalidPageCriteria`.) and generate error modal accordingly.

**Type** | **Message** | **Link formation**
------- | ------- | -------
Invalid sort criteria | Generated at `Ermrestjs` | Strip `sort` and `after, before` 
Invalid page criteria | Generated at `Ermrestjs` | Strip `after, before` 
Invalid facet | Generated at `Ermrestjs` | Strip `facet` and other conditions can co-exist
Invalid filter | Generated at `Ermrestjs` | Strip `filter` and other conditions can co-exist

Depending upon which error is being thrown the modal header changes and asks user to click on the OK button to reload the page without the faulty condition.
You can play around with the condition [here](https://dev.isrd.isi.edu/~amit/chaise1/record/#1/isa:dataset/id=269@after(id)) to see the behavior in error modal.

Minor modifications are done in `common/modal.js` and `utils.js` to accommodate modal body changes.
Test cases are written in [`errors.spec.js`](https://github.com/informatics-isi-edu/chaise/blob/facetErrors/test/e2e/specs/all-features-confirmation/errors/errors.spec.js)

### CODE REVIEW

**Comment by amitjha21** [comment]
Test cases were added. Could not test at Travis since timestamp issue is failing.

**Comment by RFSH** [comment]
@amitjha21 If you remember in the meeting we mentioned that we don't want this logic to be in catch all. You cannot assume this error is being thrown for the read request on the main entity. It should be on the catch function (or error callback) for the read request on the main entity. More specifically these reads: [recordset](https://github.com/informatics-isi-edu/chaise/blob/master/recordset/recordset.js#L318), [recordset-with-faceting](https://github.com/informatics-isi-edu/chaise/blob/master/common/table.js#L209), [recordset-re-read](https://github.com/informatics-isi-edu/chaise/blob/master/common/table.js#L161), [record](https://github.com/informatics-isi-edu/chaise/blob/master/record/record.app.js#L156), [recordedit](https://github.com/informatics-isi-edu/chaise/blob/master/recordedit/recordEdit.app.js#L232)

Also the timestamp issue is in a PR and test cases should pass. You can [see](https://travis-ci.org/informatics-isi-edu/chaise/builds/341126248#L3607) that the build in this branch is failing on new test cases that you have added.

File: .travis.yml [hunk 1]
```diff
@@ -77,6 +77,7 @@ before_install:
   - cd ..
   - git clone https://github.com/informatics-isi-edu/ermrestjs.git
   - cd ermrestjs
+  - git checkout -qf facetingErrors
```
Comment (jrchudy) [inline .travis.yml]: please remove this change.

File: common/table.js [hunk 2]
```diff
@@ -218,7 +221,14 @@
                     return defer.resolve(true);
                 }).catch(function(err) {
                     if (vm.reference.uri !== uri) {
-                        return defer.resolve(false);
+                        defer.resolve(false);
+                    } else {
```
Comment (jrchudy) [inline common/table.js]: Your else case here ignores the code that is supposed to run outside of it. This was previously refactored to behave like the following:

```
if vm.reference.uri !== uri) {
  return defer.resolve(false)
}
// continue code execution if above condition is not met
vm.hasLoaded = true;
vm.initialized = true;
...
return defer.reject(err);
```

File: common/table.js [hunk 3]
```diff
@@ -218,7 +221,14 @@
...
+                        vm.hasLoaded = true;
+                        vm.initialized = true;
+                        if(err.errorData && err.errorData.redirectPath && err.errorData.redirectPath != ''){
```
Comment (jrchudy) [inline common/table.js]: What you are doing in this `if` condition should replace `...` that I typed in the above code block.

**Pull request merged**